src/Features/CSharp/IntroduceVariable/CSharpIntroduceVariableService.cs

```diff
@@ -51,6 +51,28 @@ protected override bool IsInConstructorInitializer(ExpressionSyntax expression)
             return expression.GetAncestorOrThis<ConstructorInitializerSyntax>() != null;
         }
 
+        protected override bool IsInExpressionBodiedMember(ExpressionSyntax expression)
+        {
+            var arrowExpression = expression.FirstAncestorOrSelf<ArrowExpressionClauseSyntax>();
+            if (arrowExpression == null)
+            {
+                return false;
+            }
+
+            // If were in an expression bodied member and if the expression
+            // has a block body, then, act as if we're in a block context
+            // and not in an expression body context. 
+            // case(s)  => anonymous method expression | simple lambda expression 
+            //                                         | parenthesized lambda expression
+            var expressionBody = arrowExpression.Expression;
+            if (expressionBody is AnonymousFunctionExpressionSyntax)
+            {
+                return !((AnonymousFunctionExpressionSyntax)expressionBody).Body.IsKind(SyntaxKind.Block);
+            }
+
+            return true;
+        }
+
```

Comment:
I still don't think this is what you want.  I think instead of checking FirstAncestorOrSelf, you should find the nearest enclosing block - If you find one before you find an `ArrowExpressionClauseSyntax`, then you don't need to do anything, otherwise you do.

This would still not work if it was inside an anonymous method that was passed as an argument to a method inside an expression bodied member I think:

``` C#
public int Prop => Something.Method(delegate
    {
        // Introduce a local within this block.
    }
```